pysph/base/z_order_gpu_nnps.pyx

```diff
@@ -79,16 +81,22 @@ cdef class ZOrderGPUNNPS(GPUNNPS):
         self.domain.update()
         self.update()
 
+        self._profile = get_config().profile
+
     cpdef _bin(self, int pa_index):
         cdef NNPSParticleArrayWrapper pa_wrapper = self.pa_wrappers[pa_index]
 
         fill_pids = self.helper.get_kernel("fill_pids")
 
         pa_gpu = pa_wrapper.pa.gpu
-        fill_pids(pa_gpu.x, pa_gpu.y, pa_gpu.z,
-                self.cell_size, self.xmin[0], self.xmin[1], self.xmin[2],
+        event = fill_pids(pa_gpu.x, pa_gpu.y, pa_gpu.z,
+                self.cell_size,
+                self.make_vec(self.xmin[0], self.xmin[1], self.xmin[2]),
                 self.pid_keys[pa_index].array, self.pids[pa_index].array)
 
+        if self._profile:
```

Comment:
Instead, should we just use a decorator so we say 
```
fill_pids = profile(self.helper.get_kernel("fill_pids"))
```
and just go about calling `fill_pids`?